File: app/src/main/java/br/com/kakobotasso/testeandroidv2/currency/CurrencyRouter.java
package br.com.kakobotasso.testeandroidv2.currency;

import android.content.Intent;

import java.lang.ref.WeakReference;

import br.com.kakobotasso.testeandroidv2.login.LoginActivity;

interface CurrencyRouterInput {
    void goToLoginScreen();
}

public class CurrencyRouter implements CurrencyRouterInput {
    public WeakReference<CurrencyActivity> activity;

    @Override
    public void goToLoginScreen() {
        Intent intent = new Intent(activity.get(), LoginActivity.class);
        activity.get().startActivity(intent);
        activity.get().finish();
    }
}

File: BankApp/app/src/main/java/com/schaefer/bankapp/currency_screen/router/CurrencyRouter.kt
package com.schaefer.bankapp.login_screen.router

import android.content.Intent
import com.schaefer.bankapp.currency_screen.activity.CurrencyActivity
import java.lang.ref.WeakReference

class CurrencyRouter : CurrencyRouterInput {
    var activity: WeakReference<CurrencyActivity>? = null
    override fun passDataToNextScene(intent: Intent) {
        activity?.get()?.startActivity(intent)
        activity?.get()?.finish()
    }
}
